### PR TITLE
Exclude files listed in .bkignore

### PR DESCRIPTION
**Pull request by freeformz**

### CODE REVIEW

File: commands/publish.js [hunk 1]
```diff
@@ -47,7 +47,12 @@ function * run (context, heroku) {
   yield cli.action(`Publishing ${name} buildkit`, {success: false}, co(function * () {
     const tmpdir = tmp.dirSync().name
     const tgz = path.join(tmpdir, 'buildpack.tgz')
-    yield execa.shell(`cd ${d} && tar czf ${tgz} --exclude=.git .`)
+    var excludes = '--exclude=.git'
+    var ignore = path.join(d, '.bkignore')
+    if (fs.existsSync(ignore)) {
+      excludes += `-X ${d}`
```
Comment (tt) [inline commands/publish.js]: Should these be single quotes?

**Comment by freeformz** [comment]
FWIW: I'd like to have this in place (vs. remembering to move files out of the way) before deploying the next version of the go buildpack.

**Comment by jkutner** [comment]
I'm +1. Do you want to merge and release the plugin?

**Comment by jkutner** [comment]
Although, I do lean toward a more explicit filename, like `.buildkits-ignore`. but not deadset on it.

**Comment by freeformz** [comment]
@jkutner I don't know how to release a plugin. ;-) 

Happy to change the name of the file if that's what you want, let me know.

**Comment by edmorley** [comment]
I wonder if adding a buildpack-specific ignore file might end up being a lot more work, since now every buildpack will have to know of its existence and create their own. I've filed #24 for proposing taking `.gitignore` into account instead, plus adding more entries to this plugins default exclude list (in addition to the existing `.git` exclusion).

**Comment by edmorley** [comment]
Although that said, I guess some buildpacks may want to say exclude the tests directory from the published archive to keep its file size down (and that directory obviously wouldn't be in .gitignore).

It seems like a combined approach between this and #24 might be best.

**Pull request closed without merging**